neuralcompression/data/clic2020.py

```diff
@@ -0,0 +1,117 @@
+import os
+import pathlib
+import shutil
+import typing
+
+import PIL.Image
+import torch.utils.data
+import torchvision.datasets.folder
+import torchvision.datasets.utils
+
+
+class CLIC2020(torch.utils.data.Dataset):
+    """`Challenge on Learned Image Compression (CLIC) 2020
+    <http://compression.cc/tasks/>`_ Dataset.
+
+    Args:
+        root (str): Root directory where images are downloaded to.
+            Expects the following folder structure if download=False:
+
+            .. code::
+
+                <root>
+                    └── clic2020
+                        ├── test
+                        │   ├── *.png
+                        ├── train
+                        │   ├── *.png
+                        └── val
+                            └── *.png
+        split (str): The dataset split to use. One of
+            {``train``, ``val``, ``test``}.
+            Defaults to ``train``.
+        download (bool, optional): If true, downloads the dataset from the
+            internet and puts it in root directory. If dataset is already
+            downloaded, it is not downloaded again.
+        transform (callable, optional): A function/transform that takes in a
+            PIL image and returns a transformed version.  E.g,
+            ``transforms.RandomCrop``.
+    """
+
+    splits = {
+        "train": {
+            "filename": "train.zip",
+            "md5": "a6845cac88c3dd882246575f7a2fc5f9",
+            "url": "https://data.vision.ee.ethz.ch/cvl/clic/professional_train_2020.zip",
+        },
+        "val": {
+            "filename": "val.zip",
+            "md5": "7111ee240435911db04dbc5f40d50272",
+            "url": "https://data.vision.ee.ethz.ch/cvl/clic/professional_valid_2020.zip",
+        },
+        "test": {
+            "filename": "test.zip",
+            "md5": "4476b708ea4c492505dd70061bebe202",
+            "url": "https://storage.googleapis.com/clic2021_public/professional_test_2021.zip",
+        },
+    }
+
+    def __init__(
+        self,
+        root: typing.Union[str, pathlib.Path],
+        split: str = "train",
+        download: bool = False,
+        transform: typing.Optional[typing.Callable] = None,
```

Comment:
Could annotate further specify this as a `typing.Callable[PIL.Image, torch.Tensor]`.